src/gt4py/next/common.py

```diff
@@ -233,6 +233,14 @@ def __call__(self, func: fbuiltins.BuiltInFunction[_R, _P], /) -> Callable[_P, _
 class Field(Protocol[DimsT, core_defs.ScalarT]):
     __gt_builtin_func__: ClassVar[GTBuiltInFuncDispatcher]
 
+    @classmethod
+    def register_builtin_func(
+        cls, op: fbuiltins.BuiltInFunction[_R, _P], op_func: Optional[Callable[_P, _R]] = None
+    ) -> Callable[_P, _R] | functools.partial[Callable[_P, _R]]:
+        raise NotImplementedError
+
+    _builtin_func_map: dict[fbuiltins.BuiltInFunction, Callable] = {}
+
```

Comment:
this should not be here

Reply:
The reason I added it was for mypy as otherwise it flags the following:

```
src/gt4py/next/embedded/nd_array_field.py:256:5: error: "type[_BaseNdArrayField[Any, Any]]" has no attribute "register_builtin_func"  [attr-defined]

src/gt4py/next/common.py:488:5: error: "type[_BaseNdArrayField[Any, Any]]" has no attribute "_builtin_func_map"  [attr-defined]
```

Should I just ignore those lines then?

Reply:
I've fixed it with setting some class attributes if `TYPE_CHECKING`